test/housekeeping_test.js

```diff
@@ -323,5 +350,92 @@ describe('House Keeper', () => {
       });
     }
   });
-});
 
+  it('should call sweepVolumes exactly once', async() => {
+    houseKeeperMock.expects("sweepVolumes").exactly(regions.length);
+    
+    describeInstancesStub.returns({
+      Reservations: [
+      ]
+    });
+    
+    describeSpotInstanceRequestsStub.returns({
+      SpotInstanceRequests: [
+      ]
+    });
+    
+    await houseKeeper.sweep();
+    houseKeeperMock.verify();
+  });
+
+  it('should not fail if no volume data is returned', async() => {
+    houseKeeperMock.expects("handleVolumeData").never();
+    describeInstancesStub.returns({
+      Reservations: [
+      ]
+    });
+    
+    describeSpotInstanceRequestsStub.returns({
+      SpotInstanceRequests: [
+      ]
+    });
+    
+    describeVolumesStub.returns({
+      Volumes: [
+      ]
+    });
+
+    await houseKeeper.sweep();
+    houseKeeperMock.verify();
+  });
+  
+  it('should call handleVolumeData once per volume', async() => {
+    houseKeeperMock.expects("handleVolumeData").twice();
+      
+    describeInstancesStub.returns({
+      Reservations: [
+      ]
+    });
+    
+    describeSpotInstanceRequestsStub.returns({
+      SpotInstanceRequests: [
+      ]
+    });
+
+    describeVolumesStub.withArgs(ec2[region], 'describeVolumes', houseKeeper.describeVolumesParams)
+      .returns({
+        Volumes: [{
+          Attachments: [],
+          AvailabilityZone: region, 
+          CreateTime: new Date().toString(), 
+          Size: 8, 
+          SnapshotId: "snap-1234567890abcdef0", 
+          State: "in-use", 
+          VolumeId: "vol-049df61146c4d7901", 
+          VolumeType: "standard",
+        }]
+      });
+
+    describeVolumesStub.withArgs(ec2['us-east-2'], 'describeVolumes', houseKeeper.describeVolumesParams)
```

Comment:
I'm curious why you chose to use `ec2[region]` for the first call, but a specific region for the second one?  I wonder if we should either pick two hard coded regions or do something like `Object.keys(ec2)[0]` and `Object.keys(ec2)[1]`?

Reply:
That's a great point. We wanted to use the global `region` since it was already defined, but wanted to include a second region in the test without creating another global variable. In this case, it would probably make more sense to use two hard-coded keys, since we need both the region string and the corresponding ec2 instance.